server/http/dialog/install.go

```diff
@@ -128,23 +142,26 @@ func (d *dialog) handleInstall(w http.ResponseWriter, req *http.Request) {
 	v = dialogRequest.Submission["secret"]
 	secret, _ := v.(string)
 
-	var manifest apps.Manifest
-	err = json.Unmarshal([]byte(dialogRequest.State), &manifest)
+	manifest, err := d.apps.AppClient.GetManifest(dialogRequest.State)
```

Comment:
I still prefer to have the whole manifest in the state (whether it is a KV record or on the dialog state). Use case:
- User wants to install App
- Retrieve manifest
- While user is reading the permissions, the app updates its manifest to ask for more permissions
- User accepts the permissions shown in the dialog
- We refetch the manifest from the app, which give us a manifest with more permissions than what we show to the user
- The app gets installed with permissions the user never accepted

Doesn't need to be covered in this PR, but we have to keep this in mind.

Reply:
Here's the story behind this.

1. I agree. I struggled to squeeze everything into the Interactive Dialog structs, and this is where I ended up after a couple iterations. Need to think some more, may switch back.
2. Much more importantly. I sincerely hope that there will be few or no "old-style"  Interactive Dialogs in this code base once we are done. After we build `Modal`s, we might as well use them in this plugin itself. That's my hope - its UI should become that of an App.